### ISSUE TITLE
How to use?

### ISSUE THREAD

**Issue by dineshraj01** [sumefsp/LPM-Algorithms-Linux-XIA#1]
How can one Test the algorithms? A little documentation would help.

**Comment by sarvghotra** [comment]
@sumefsp Could you please help to regenerate the results by giving some pointers to run this code ?

**Comment by garnaiksumeet** [comment]
@sarvjeetsinghghotra Run evaluate.sh to generate the graph for the algorithms tested i.e bloom and radix. For test of correctness, run correctness.sh.

**Comment by sarvghotra** [comment]
@sumefsp Thank you so much for a quick reply. 

I had some how already figured that out. But evaluate.sh only compares boom and radix. I want to evaluate proptrie also (either with them or seperately). So could you please help with the poptrie ? (Even the original implementation page doesn't have instruction to run, at posix/poptrie).

**Comment by garnaiksumeet** [comment]
@sarvjeetsinghghotra Yes, you are correct that evaluate.sh only compares bloom and radix since, those are verified for the correctness of the implementation. The poptrie code in this repo is incomplete. It is not yet implemented for XIA in this project. You could take a look at https://github.com/gvaibhavraj2/poptrie-XID/tree/final. I am not sure of the implementation in the other repo. You could look around the XIA google group for more info regarding that project. If it is fully implemented, just give another comment and I will add a page on how to integrate newer algorithms.

**Comment by sarvghotra** [comment]
I have figured out to evaluate the poptrie using the original implementation given here. https://github.com/pixos/poptrie

I have read the poptrie paper but could not understand what is Radix tree used for in its implementation. Where as in the paper Radix tree is not mentioned. So could you please explain the function of radix tree ?

Thanks in advance.
